Makefile

```diff
@@ -1,181 +1,203 @@
+	cd examples/kafka-plugin/manual-partitioner && go build -ldflags "${LDFLAGS}" ${GO_BUILD_ARGS}
+	cd examples/kafka-plugin/post-init-consumer && go build -ldflags "${LDFLAGS}" ${GO_BUILD_ARGS}
+	cd examples/logs-plugin                     && go build -ldflags "${LDFLAGS}" ${GO_BUILD_ARGS}
+	cd examples/redis-plugin                    && go build -ldflags "${LDFLAGS}" ${GO_BUILD_ARGS}
+	cd examples/simple-agent                    && go build -ldflags "${LDFLAGS}" ${GO_BUILD_ARGS}
+	cd examples/statuscheck-plugin              && go build -ldflags "${LDFLAGS}" ${GO_BUILD_ARGS}
+	cd examples/prometheus-plugin               && go build -ldflags "${LDFLAGS}" ${GO_BUILD_ARGS}
+	cd examples/cryptodata-plugin               && go build -ldflags "${LDFLAGS}" ${GO_BUILD_ARGS}
+	cd examples/bolt-plugin                     && go build -ldflags "${LDFLAGS}" ${GO_BUILD_ARGS}
+
 clean-examples:
-	@echo "=> cleaning examples"
-	cd examples/cassandra-lib && rm -f cassandra-lib
-	cd examples/etcd-lib && make clean
-	cd examples/kafka-lib && make clean
-	cd examples/logs-lib && make clean
-	cd examples/redis-lib && make clean
-
-# Clean plugin examples
+	@echo "# cleaning examples"
+	cd examples/cassandra-lib  && go clean
+	cd examples/etcd-lib       && make clean
+	cd examples/kafka-lib      && make clean
+	cd examples/logs-lib       && make clean
+	cd examples/redis-lib      && make clean
+	cd examples/cryptodata-lib && go clean
+
 clean-examples-plugin:
-	@echo "=> cleaning plugin examples"
-	rm -f examples/configs-plugin/configs-plugin
-	rm -f examples/datasync-plugin/datasync-plugin
-	rm -f examples/flags-lib/flags-lib
-	rm -f examples/kafka-plugin/hash-partitioner/hash-partitioner
-	rm -f examples/kafka-plugin/manual-partitioner/manual-partitioner
-	rm -f examples/kafka-plugin/post-init-consumer/post-init-consumer
-	rm -f examples/logs-plugin/logs-plugin
-	rm -f examples/redis-plugin/redis-plugin
-	rm -f examples/simple-agent/simple-agent
-	rm -f examples/statuscheck-plugin/statuscheck-plugin
-	rm -f examples/prometheus-plugin/prometheus-plugin
-	rm -f examples/bolt-plugin/bolt-plugin
-
-# Get test tools
-get-testtools:
-	@echo "=> installing test tools"
+	@echo "# cleaning plugin examples"
+	cd examples/configs-plugin                  && go clean
+	cd examples/datasync-plugin                 && go clean
+	cd examples/flags-lib                       && go clean
+	cd examples/kafka-plugin/hash-partitioner   && go clean
+	cd examples/kafka-plugin/manual-partitioner && go clean
+	cd examples/kafka-plugin/post-init-consumer && go clean
+	cd examples/logs-plugin                     && go clean
+	cd examples/redis-plugin                    && go clean
+	cd examples/simple-agent                    && go clean
+	cd examples/statuscheck-plugin              && go clean
+	cd examples/prometheus-plugin               && go clean
+	cd examples/bolt-plugin                     && go clean
```

Comment:
looks like `examples/cryptodata-plugin` missing

Reply:
maybe where is a way to `forloop` it

Reply:
Actually, we could use `go list` to list all Go programs (`main` packages) under `examples` directory:

```
➤ go list -f '{{if eq .Name "main"}}{{.Dir}}{{end}}' ./examples/...
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/bolt-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/cassandra-lib
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/configs-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/consul-lib
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-lib
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-proto-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/datasync-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/editor
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/election
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/view
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/watcher
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/flags-lib
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/grpc-plugin/grpc-client
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/grpc-plugin/grpc-server
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/asyncproducer
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/consumer
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/mux
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/syncproducer
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/hash-partitioner
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/manual-partitioner
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/post-init-consumer
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/basic
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/custom
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/http
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/advanced-scenario
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/basic-scenario
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/templates
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/test-process
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/prometheus-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-lib/airport
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-lib/simple
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/simple-agent
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/statuscheck-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/supervisor-plugin
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/01_hello-world
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/02_plugin-deps
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/03_rest-handler
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/04_kv-store
/home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/06_plugin_lookup
```

And then we can actually pipe this output into `go clean` using `xargs`:

```
➤ go list -f '{{if eq .Name "main"}}{{.Dir}}{{end}}' ./examples/... | xargs go clean -x
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/bolt-plugin
rm -f bolt-plugin bolt-plugin.exe bolt-plugin.test bolt-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/cassandra-lib
rm -f cassandra-lib cassandra-lib.exe cassandra-lib.test cassandra-lib.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/configs-plugin
rm -f configs-plugin configs-plugin.exe configs-plugin.test configs-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/consul-lib
rm -f consul-lib consul-lib.exe consul-lib.test consul-lib.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-lib
rm -f cryptodata-lib cryptodata-lib.exe cryptodata-lib.test cryptodata-lib.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-plugin
rm -f cryptodata-plugin cryptodata-plugin.exe cryptodata-plugin.test cryptodata-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/cryptodata-proto-plugin
rm -f cryptodata-proto-plugin cryptodata-proto-plugin.exe cryptodata-proto-plugin.test cryptodata-proto-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/datasync-plugin
rm -f datasync-plugin datasync-plugin.exe datasync-plugin.test datasync-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/editor
rm -f editor editor.exe editor.test editor.test.exe doc doc.exe editor editor.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/election
rm -f election election.exe election.test election.test.exe election election.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/view
rm -f view view.exe view.test view.test.exe doc doc.exe view view.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/etcd-lib/watcher
rm -f watcher watcher.exe watcher.test watcher.test.exe doc doc.exe watcher watcher.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/flags-lib
rm -f flags-lib flags-lib.exe flags-lib.test flags-lib.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/grpc-plugin/grpc-client
rm -f grpc-client grpc-client.exe grpc-client.test grpc-client.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/grpc-plugin/grpc-server
rm -f grpc-server grpc-server.exe grpc-server.test grpc-server.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/asyncproducer
rm -f asyncproducer asyncproducer.exe asyncproducer.test asyncproducer.test.exe asyncproducer asyncproducer.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/consumer
rm -f consumer consumer.exe consumer.test consumer.test.exe consumer consumer.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/mux
rm -f mux mux.exe mux.test mux.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-lib/syncproducer
rm -f syncproducer syncproducer.exe syncproducer.test syncproducer.test.exe doc doc.exe syncproducer syncproducer.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/hash-partitioner
rm -f hash-partitioner hash-partitioner.exe hash-partitioner.test hash-partitioner.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/manual-partitioner
rm -f manual-partitioner manual-partitioner.exe manual-partitioner.test manual-partitioner.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/kafka-plugin/post-init-consumer
rm -f post-init-consumer post-init-consumer.exe post-init-consumer.test post-init-consumer.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/basic
rm -f basic basic.exe basic.test basic.test.exe basic basic.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/custom
rm -f custom custom.exe custom.test custom.test.exe custom custom.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-lib/http
rm -f http http.exe http.test http.test.exe doc doc.exe server server.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/logs-plugin
rm -f logs-plugin logs-plugin.exe logs-plugin.test logs-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/advanced-scenario
rm -f advanced-scenario advanced-scenario.exe advanced-scenario.test advanced-scenario.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/basic-scenario
rm -f basic-scenario basic-scenario.exe basic-scenario.test basic-scenario.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/templates
rm -f templates templates.exe templates.test templates.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/process-manager-plugin/test-process
rm -f test-process test-process.exe test-process.test test-process.test.exe test-process test-process.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/prometheus-plugin
rm -f prometheus-plugin prometheus-plugin.exe prometheus-plugin.test prometheus-plugin.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-lib/airport
rm -f airport airport.exe airport.test airport.test.exe airport airport.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-lib/simple
rm -f simple simple.exe simple.test simple.test.exe doc doc.exe simple simple.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/redis-plugin
rm -f redis-plugin redis-plugin.exe redis-plugin.test redis-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/simple-agent
rm -f simple-agent simple-agent.exe simple-agent.test simple-agent.test.exe agent agent.exe doc doc.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/statuscheck-plugin
rm -f statuscheck-plugin statuscheck-plugin.exe statuscheck-plugin.test statuscheck-plugin.test.exe doc doc.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/supervisor-plugin
rm -f supervisor-plugin supervisor-plugin.exe supervisor-plugin.test supervisor-plugin.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/01_hello-world
rm -f 01_hello-world 01_hello-world.exe 01_hello-world.test 01_hello-world.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/02_plugin-deps
rm -f 02_plugin-deps 02_plugin-deps.exe 02_plugin-deps.test 02_plugin-deps.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/03_rest-handler
rm -f 03_rest-handler 03_rest-handler.exe 03_rest-handler.test 03_rest-handler.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/04_kv-store
rm -f 04_kv-store 04_kv-store.exe 04_kv-store.test 04_kv-store.test.exe main main.exe
cd /home/ondrej/go/src/github.com/ligato/cn-infra/examples/tutorials/06_plugin_lookup
rm -f 06_plugin_lookup 06_plugin_lookup.exe 06_plugin_lookup.test 06_plugin_lookup.test.exe main main.exe plugin1 plugin1.exe plugin2 plugin2.exe
```

This seems most universal way to do it.